source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/input/TextInputRenderer.java

```diff
@@ -152,6 +187,79 @@ public void afterTextChanged(Editable s)
             }
         });
 
+        TextInput textInput = (TextInput)baseInputElement;
+        BaseActionElement action = textInput.GetInlineAction();
+
+        LinearLayout textInputViewGroup = null;
+        if(action != null)
+        {
+            if (hostConfig.GetActions().getShowCard().getActionMode() == ActionMode.Inline &&
+                (action.GetElementType() == ActionType.ShowCard))
+            {
+                renderedCard.addWarning(new AdaptiveWarning(AdaptiveWarning.INTERACTIVITY_DISALLOWED, "Inline ShowCard not supported for InlineAction"));
+            }
+            else
+            {
+                textInputViewGroup = new LinearLayout(context);
```

Comment:
>textInputViewGroup = new LinearLayout(context); [](start = 16, length = 47)

[https://stackoverflow.com/questions/32723143/android-layout-with-match-parent-shall-take-the-space-left-or-share](https://stackoverflow.com/questions/32723143/android-layout-with-match-parent-shall-take-the-space-left-or-share)

Have you taken a look at the first answer? it states that you can use a relative layout as the first view grabs all the space before the button has any chance to grab any

Reply:
adjusted weight to make layout clearer

---
In reply to: [234329614](https://github.com/Microsoft/AdaptiveCards/pull/2167#discussion_r234329614) [](ancestors = 234329614)